internal/services/amlfilesystem/aml_file_system_resource.go

```diff
@@ -0,0 +1,518 @@
+	maintenanceWindow := &input[0]
+
+	return amlfilesystems.AmlFilesystemPropertiesMaintenanceWindow{
+		DayOfWeek:    pointer.To(maintenanceWindow.DayOfWeek),
+		TimeOfDayUTC: pointer.To(maintenanceWindow.TimeOfDayInUTC),
+	}
+}
+
+func expandAMLFileSystemMaintenanceWindowForUpdate(input []MaintenanceWindow) *amlfilesystems.AmlFilesystemUpdatePropertiesMaintenanceWindow {
+	if len(input) == 0 {
+		return nil
+	}
+
+	maintenanceWindow := &input[0]
+	result := amlfilesystems.AmlFilesystemUpdatePropertiesMaintenanceWindow{}
+
+	if v := maintenanceWindow.DayOfWeek; v != "" {
+		result.DayOfWeek = pointer.To(v)
+	}
+
+	if v := maintenanceWindow.TimeOfDayInUTC; v != "" {
+		result.TimeOfDayUTC = pointer.To(v)
+	}
+
+	return &result
+}
+
+func flattenAMLFileSystemMaintenanceWindow(input amlfilesystems.AmlFilesystemPropertiesMaintenanceWindow) []MaintenanceWindow {
+	var result []MaintenanceWindow
+
+	maintenanceWindow := MaintenanceWindow{
+		DayOfWeek:      pointer.From(input.DayOfWeek),
+		TimeOfDayInUTC: pointer.From(input.TimeOfDayUTC),
+	}
+
+	return append(result, maintenanceWindow)
+}
+
+func expandAMLFileSystemKeyEncryptionKey(input []KeyEncryptionKey) *amlfilesystems.AmlFilesystemEncryptionSettings {
+	if len(input) == 0 {
+		return nil
+	}
+
+	keyEncryptionKey := &input[0]
+
+	result := &amlfilesystems.KeyVaultKeyReference{
+		KeyUrl: keyEncryptionKey.KeyUrl,
+		SourceVault: amlfilesystems.KeyVaultKeyReferenceSourceVault{
+			Id: pointer.To(keyEncryptionKey.SourceVaultId),
+		},
+	}
+
+	return &amlfilesystems.AmlFilesystemEncryptionSettings{
+		KeyEncryptionKey: result,
+	}
+}
+
+func flattenAMLFileSystemKeyEncryptionKey(input *amlfilesystems.AmlFilesystemEncryptionSettings) []KeyEncryptionKey {
+	if input == nil || input.KeyEncryptionKey == nil {
+		return nil
+	}
+
+	keyEncryptionKey := input.KeyEncryptionKey
+
+	result := KeyEncryptionKey{
+		KeyUrl:        keyEncryptionKey.KeyUrl,
+		SourceVaultId: pointer.From(keyEncryptionKey.SourceVault.Id),
+	}
+
+	return []KeyEncryptionKey{result}
+}
+
+func expandAMLFileSystemHsmSetting(input []HsmSetting) *amlfilesystems.AmlFilesystemPropertiesHsm {
+	if len(input) == 0 {
+		return nil
+	}
+
+	hsmSetting := &input[0]
+
+	result := &amlfilesystems.AmlFilesystemHsmSettings{
+		Container:        hsmSetting.ContainerId,
+		LoggingContainer: hsmSetting.LoggingContainerId,
+	}
+
+	if hsmSetting.ImportPrefix != "" {
+		result.ImportPrefix = pointer.To(hsmSetting.ImportPrefix)
+	}
+
+	return &amlfilesystems.AmlFilesystemPropertiesHsm{
+		Settings: result,
+	}
+}
+
+func flattenAMLFileSystemHsmSetting(input *amlfilesystems.AmlFilesystemPropertiesHsm) []HsmSetting {
+	if input == nil || input.Settings == nil {
+		return nil
+	}
+
+	var result []HsmSetting
+	hsmSetting := HsmSetting{}
+
+	if v := input.Settings; v != nil {
+		hsmSetting.ContainerId = v.Container
+		hsmSetting.LoggingContainerId = v.LoggingContainer
+
+		if v.ImportPrefix != nil {
```

Comment:
Can remove this nil check as it's been properly handled in `pointer.From()`?

Reply:
Updated